File: internal/service/llm_client.go
package service

import "context"

// LLMClient представляет интерфейс для работы с различными LLM провайдерами.
// Это позволяет использовать разные провайдеры (OpenRouter, ProxyAPI и т.д.)
// через единый интерфейс.
type LLMClient interface {
	// GenerateQuestion генерирует ответ на основе промпта.
	// Возвращает текстовый ответ от LLM или ошибку.
	GenerateQuestion(ctx context.Context, prompt string) (string, error)
}

File: backend/content-service/app/clients/llm_client.py
"""LLM client interface."""

from typing import Protocol

from app.models.llm_response import LLMQuestionResponse, LLMQuestionsResponse


class LLMClient(Protocol):
    """Protocol for LLM clients."""

    async def generate_question(self, prompt: str) -> LLMQuestionResponse:
        """
        Generate question response from LLM.

        Args:
            prompt: Prompt text for LLM

        Returns:
            LLMQuestionResponse with generated question data

        Raises:
            Exception: If LLM request fails
        """
        ...

    async def generate_questions(self, prompt: str) -> LLMQuestionsResponse:
        """
        Generate multiple questions response from LLM.

        Args:
            prompt: Prompt text for LLM

        Returns:
            LLMQuestionsResponse with generated questions data

        Raises:
            Exception: If LLM request fails
        """
        ...
